### ISSUE TITLE
Orca is too chatty when switching tabs due to aria-label being redundant to title

### ISSUE THREAD

**Issue by joanmarie** [microsoft/vscode#95378]
Step to Reproduce:

1. Switch between open documents with Alt+n (where n is the index of the tab)

Orca's presentation for a file named "hello world" is:

    hello world. page tab /path/to/hello world

The reason why is that Orca by default presents (at least) the following information for a widget:

* accessible name (e.g. the value of `aria-label`)
* accessible role (e.g. the localized value of ARIA's `role`)
* accessible description (e.g. the value of `aria-description`, `aria-describedby`, or the `title`)

From the inspector, I see:

```
<div ... role="tab" class="tab ... aria-label="hello world" title="/home/jd/hello world" ...>
```

I get that knowing the path is of value and makes sense as something which appears when the user hovers the mouse over the tab. By the same token, I don't think this amount of verbosity is great for the screen reader user.

And yes, I believe I said in some other issue that putting the path where it would be exposed as the accessible description made sense because users could interrupt speech, etc. But in this particular case, user interruption might not be ideal because what Orca says immediately after is relevant to the user, namely: That the editor has focus and the current line.

I'd be interested in hearing what users (@jvesouza, @webczat, et al.) and @isidorn think.

**Comment by webczat** [comment]
For me that is not really a problem. I probably have some issues with things being too chatty, for example when pressing numpad enter and I want to read current line of text I am on... but not when switching editors.

**Comment by isidorn** [comment]
I think reading the path to a file when a tab gets focused is too verbose. There are other ways to figure out the path of the file and imho it should not be read out every time a user focuses on a file.

I just tried VoiceOver and it nicely ignores the `title` attribute and only reads the file name.

imho in general when `aria-label` is present screen readers should ignore `title` as the `aria-label` should containt the fully description. `title` should only be used when `aria-label` is not present as a substitute. Though you would know better, this just makes sense from my standpoint.

**Comment by joanmarie** [comment]
@isidorn Fair enough. But here's the thing:

1. accessible name != accessible description. See under https://w3c.github.io/aria/#terms. As you'll see there:

    > An accessible description provides additional information, related to an interface element, that complements the accessible name. 

2. For most elements when there is an `aria-label` and a `title` attribute (and no other source of name/description info), the value of `aria-label` gets exposed to platforms as the accessible name; the `title` attribute gets exposed to platforms as the accessible description. Given the definition/explanation I quoted above, screen readers that include the accessible description by default are operating under the assumption that the accessible description is providing additional information rather than redundant information.

3.  Screen readers using their accessibility APIs to get the accessible name and the accessible description *don't necessarily know what authoring caused those accessible properties to be there*. If the description was put there by, for instance, the new `aria-description` property, it would look the same to me as if it came from the `title`.

I suppose I can add a heuristic to Orca to handle this particular case (description looks like a file path and the filename seems to be in the accessible name)....

**Comment by isidorn** [comment]
Ok thanks for clarifying. I oversimplified things.

So for me this more feels like a screen reader verbosity setting. If the screen reader is set to be very concise it should only read `aria-description` when explicitly asked for by the user.
If the screen reader is in a very verbose mode than it should always first read `aria-label` then `aria-description`.

And after reading the docs I take back what I said that the `aria-label` is a substitue for the `title`. That is incorrect, however `aria-description` is a substitue, thus if `aria-description` is set I think you should ignore the `title`.

That heuristic you propose is imho a bit too specific and sounds like a hack.
A heuristic which imho makes sense is to do some string matching is see how similar the name and description are. But even then this feels a bit magical. I still think this shuold be user configured via verbostiy setting and there shuold be an explicit action to get the description being read.

**Comment by joanmarie** [comment]
Orca already has a verbosity setting for description so it can be disabled. Orca also has a command to get detailed information for the current object. But the user is only very briefly on the page tab because focus immediately goes to the editor.

And I agree about sounds like a hack. Orca already has simple comparisons for name and description, but it's not yet catching this particular case.

**Comment by isidorn** [comment]
Ok makes perfect sense.
Then I think the only way to fix this if you want is to fine tune that simple comparision which you already do. 

💡 or even better, vscode sets the `aria-description` to be an empty string `""`. `aria-descirption` takes precendence over title and orca does not read anything 💡

**Comment by joanmarie** [comment]
To be honest, I'm not 100% sure that would work. Just to be sure, I asked in the [associated PR](https://github.com/w3c/accname/pull/69#issuecomment-614738513).

**Comment by isidorn** [comment]
@joanmarie ok thanks! Let me know what they say and if we should do the `aria-descirption` change on the vscode side.
For me it feels like the client should have some way to tell the screen reader that the `title` should be ignored.

**Comment by isidorn** [comment]
Just tried out setting aria-description to an empty string and it does not work. Orca is still too chatty, but in theory if that is the right thing to do, we can do that change and you can fix it on the Orca side. So let's wait for the reply on the PR. Thanks!

**Comment by joanmarie** [comment]
So the comparison stuff I have locally (not yet committed) solves the original problem.

BUT it seems like now I'm sometimes (but not always) seeing a selection event for the `tab` _after_ the editor claims focus. This is a new development and leads to different chattiness (which sounds very similar to the chattiness I reported here).

I tried to filter out this new chattiness, but because sometimes the events are missing the filtering resulted in Orca saying nothing. I'll get to the bottom of that in a bit (hopefully!) and then open a new issue if appropriate.

**Comment by joanmarie** [comment]
So Aaron just responded. We should get an empty description. I can confirm that's the case by loading this simple test in chromium built locally: `data:text/html,<input aria-label="enter stuff here" title="tooltip!" aria-description="">` So I guess it's just a matter of VSCode / Electron using a more recent version of Chromium.

**Comment by isidorn** [comment]
@joanmarie fantastic. I have just pushed a change that we set the aria-description on the tab area to empty.
Next milestone we should update to Chrome 80 so hopefully it should just work then.

We should have this `aria-description` in mind in case more situations appear where Orca is reading a `title` and it should not.
